### PR TITLE
JSON API v2

### PR DESCRIPTION
**Pull request by TMiguelT**
A slightly altered version of #87. The main difference is that we use [`flask-rest-jsonapi`](https://github.com/miLibris/flask-rest-jsonapi) (or, [my fork of it](https://github.com/TMiguelT/flapison) at least), in order to get things like pagination, the `include` query parameter, and generally automated endpoints.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -32,26 +32,38 @@
...
+    "scipy>=1.3.1",
+    "flatten_json>=0.1.7",
+    "flapison>=0.30.1",
 
     # Testing
-    "pytest>=3.0.6",
-    "WebTest>=2.0.26",
-    "factory-boy>=2.8.1",
-    "livereload>=2.5.1",
+    "pytest==3.0.6",
```
Comment (ewels) [inline setup.py]: Do these have to be `==`? How come?

**Comment by ewels** [comment]
I'm struggling to run this code locally. I re-run `python setup.py develop` and it seems to install all of the new dependencies properly. But when I try either `megaqc run` or `megaqc initdb` I get the following:

```python
Traceback (most recent call last):
  File "/Users/ewels/miniconda3/envs/py3/bin/megaqc", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/ewels/GitHub/MegaQC/scripts/megaqc", line 39, in <module>
    cli()
  File "/Users/ewels/miniconda3/envs/py3/lib/python3.7/site-packages/Click-7.0-py3.7.egg/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/Users/ewels/miniconda3/envs/py3/lib/python3.7/site-packages/Flask-1.0.2-py3.7.egg/flask/cli.py", line 557, in main
    return super(FlaskGroup, self).main(*args, **kwargs)
  File "/Users/ewels/miniconda3/envs/py3/lib/python3.7/site-packages/Click-7.0-py3.7.egg/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/Users/ewels/miniconda3/envs/py3/lib/python3.7/site-packages/Click-7.0-py3.7.egg/click/core.py", line 1132, in invoke
    cmd_name, cmd, args = self.resolve_command(ctx, args)
  File "/Users/ewels/miniconda3/envs/py3/lib/python3.7/site-packages/Click-7.0-py3.7.egg/click/core.py", line 1171, in resolve_command
    cmd = self.get_command(ctx, cmd_name)
  File "/Users/ewels/miniconda3/envs/py3/lib/python3.7/site-packages/Flask-1.0.2-py3.7.egg/flask/cli.py", line 515, in get_command
    rv = info.load_app().cli.get_command(ctx, name)
  File "/Users/ewels/miniconda3/envs/py3/lib/python3.7/site-packages/Flask-1.0.2-py3.7.egg/flask/cli.py", line 367, in load_app
    app = call_factory(self, self.create_app)
  File "/Users/ewels/miniconda3/envs/py3/lib/python3.7/site-packages/Flask-1.0.2-py3.7.egg/flask/cli.py", line 114, in call_factory
    return app_factory(script_info)
  File "/Users/ewels/GitHub/MegaQC/scripts/megaqc", line 12, in create_megaqc_app
    from megaqc.app import create_app
  File "/Users/ewels/GitHub/MegaQC/megaqc/app.py", line 14, in <module>
    from megaqc import commands, public, user, version, api, rest_api
  File "/Users/ewels/GitHub/MegaQC/megaqc/rest_api/__init__.py", line 1, in <module>
    from . import filters, schemas, utils, views
  File "/Users/ewels/GitHub/MegaQC/megaqc/rest_api/views.py", line 385, in <module>
    class TrendSeries(ResourceList):
  File "/Users/ewels/GitHub/MegaQC/megaqc/rest_api/views.py", line 388, in TrendSeries
    @use_kwargs(schemas.TrendInputSchema(), locations=("querystring",))
  File "/Users/ewels/miniconda3/envs/py3/lib/python3.7/site-packages/webargs-6.0.0b2-py3.7.egg/webargs/core.py", line 386, in use_kwargs
    return self.use_args(*args, **kwargs)
TypeError: use_args() got an unexpected keyword argument 'locations'
```

**Comment by ewels** [comment]
NB: I figured it was something odd in my environment when I first hit this earlier today, but I just tried on my machine at home at got the same error..

File: setup.py (hunk 1 above)
Comment (TMiguelT) [reply setup.py]: No reason, good catch

**Comment by TMiguelT** [comment]
This bug you're getting is because I used fairly loose requirements specifiers in the setup.py (`>=`), and a new major release of webargs came out, which my code is incompatible with. I changed them to `~=`, which will never jump a major version, so this shouldn't recur.

**Comment by ewels** [comment]
ok brilliant, thanks for the fix! That did the job..

**Pull request merged**